File: java_solutions/1422.maximum-score-after-splitting-a-string.java
/*
 * @lc app=leetcode id=1422 lang=java
 *
 * [1422] Maximum Score After Splitting a String
 */

class Solution {
    public int maxScore(String s) {
        return IntStream.range(1, s.length())
            .map(i -> (int) s.substring(0, i).chars().filter(c -> c == '0').count() 
                      + (int) s.substring(i).chars().filter(c -> c == '1').count())
            .max()
            .orElse(0);
    }
}

File: kotlin_solutions_1/1422.maximum-score-after-splitting-a-string.kt
/*
 * @lc app=leetcode id=1422 lang=kotlin
 *
 * [1422] Maximum Score After Splitting a String
 */

class Solution { fun maxScore(s: String): Int = (1 until s.length).maxOf { i -> s.substring(0, i).count { it == '0' } + s.substring(i).count { it == '1' } } }